include/seqan3/core/algorithm/configuration_utility.hpp

```diff
@@ -0,0 +1,153 @@
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+// LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+// OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+// DAMAGE.
+//
+// ============================================================================
+
+/*!\file
+ * \brief Provides functionality to access get function by enum values.
+ * \author Rene Rahn <rene.rahn AT fu-berlin.de>
+ */
+
+#pragma once
+
+#include <array>
+#include <type_traits>
+
+#include <meta/meta.hpp>
+
+#include <seqan3/core/algorithm/concept.hpp>
+#include <seqan3/core/type_list.hpp>
+#include <seqan3/core/concept/tuple.hpp>
+
+namespace seqan3::detail
+{
+
+// ----------------------------------------------------------------------------
+// compatibility_table
+// ----------------------------------------------------------------------------
+
+/*!\brief Declaration of algorithm specific compatibility table.
+ * \ingroup algorithm
+ * \tparam algorithm_id_type The type of the algorithm specific id. Algorithm configurations must maintain
+ *                           this table to allow validation checks.
+ */
+template <typename algorithm_id_type>
+inline constexpr std::array<std::array<void, 0>, 0> compatibility_table;
+
+// ----------------------------------------------------------------------------
+// Metafunction is_configuration_valid
+// ----------------------------------------------------------------------------
+
+/*!\brief Value metafunction which checks if a given type is compatible with a list of other types.
+ * \ingroup algorithm
+ * \tparam query_t       The type to check for compatibility.
+ * \tparam compare_types The types to compare against.
+ *
+ * \details
+ *
+ * Checks if the type is from the same algorithm configuration and if it can be combined with any of the
+ * existing elements in the current configuration.
+ *
+ * \see seqan3::detail::is_configuration_valid_v
+ */
+template <config_element_concept query_t, config_element_concept ... compare_types>
+struct is_configuration_valid
+{
+    //!\privatesection
+    //!\brief Helper typedef.
+    using algorithm_id_type = remove_cvref_t<decltype(query_t::id)>;
+    //!\brief Helper typedef.
+    using index_type        = std::underlying_type_t<algorithm_id_type>;
+
+    //!\publicsection
+    //!\brief The result of checking the compatibility.
+    static constexpr bool value =
+        (std::is_same_v<algorithm_id_type, remove_cvref_t<decltype(compare_types::id)>> && ...) &&
+        (compatibility_table<algorithm_id_type>[static_cast<index_type>(query_t::id)]
+                                             [static_cast<index_type>(compare_types::id)] && ...);
+};
```

Comment:
I realise this would make reading it a lot more difficult, but we said that we want to make the value metafunction inherit std::integral_constat... can you refactor this a little so it works?